qemu/tests/driver_load.py

```diff
@@ -70,24 +78,28 @@ def check_driver(session, cmd, pattern):
 
     devcon = params.get("devcon")
     if devcon:
-        error.context("Copy devcon.exe from winutils.iso to C:\\")
+        error_context.context("Copy devcon.exe from winutils.iso to C:\\")
         copy_devcon_cmd = params.get("devcon") % \
             utils_misc.get_winutils_vol(session)
         session.cmd(copy_devcon_cmd)
 
     for repeat in range(0, int(params.get("repeats", 1))):
-        error.context("Unload and load the driver. Round %s" % repeat,
-                      logging.info)
-        error.context("Get driver info from guest", logging.info)
+        error_context.context("Unload and load the driver. Round %s" % repeat,
+                              logging.info)
+        error_context.context("Get driver info from guest", logging.info)
         driver_id = check_driver(session, driver_id_cmd, driver_id_pattern)
 
-        error.context("Unload the driver", logging.info)
-        unload_driver(session, driver_unload_cmd, driver_id)
-        time.sleep(5)
-
-        error.context("Load the driver", logging.info)
-        load_driver(session, driver_load_cmd, driver_id)
-        time.sleep(5)
+        try:
+            error_context.context("Unload the driver", logging.info)
+            unload_driver(session, driver_unload_cmd, driver_id)
+            session = vm.wait_for_login(nic_index=nic_index, timeout=timeout)
+            time.sleep(5)
+        except exceptions.TestFail:
+            raise
+        finally:
+            error_context.context("Load the driver", logging.info)
+            load_driver(session, driver_load_cmd, driver_id)
+            time.sleep(5)
```

Comment:
@suqinhuang 

According to the request, need run rng again after enable rng driver.
I'm not sure whether what's the final decision about it.